src/openrct2-ui/windows/ServerList.cpp

```diff
@@ -267,22 +269,17 @@ class ServerListWindow final : public Window
         if (text.empty())
             return;
 
-        std::string temp = static_cast<std::string>(text);
+        auto temp = u8string{ text };
 
         switch (widgetIndex)
         {
             case WIDX_PLAYER_NAME_INPUT:
                 if (_playerName == text)
                     return;
 
-                text.copy(_playerName, sizeof(_playerName));
-
-                // Don't allow empty player names
-                if (_playerName[0] != '\0')
-                {
-                    gConfigNetwork.PlayerName = _playerName;
-                    ConfigSaveDefault();
-                }
+                _playerName = temp;
+                gConfigNetwork.PlayerName = _playerName;
+                window_server_list_widgets[WIDX_PLAYER_NAME_INPUT].string = const_cast<utf8*>(_playerName.c_str());
```

Comment:
Since the buffer is also set here, I suppose that it's safe, but it's still bad design. One way to ensure that the buffer remains valid is to not use u8string's internal buffer directly.

Since the current implementation works, feel free to just leave this for now.

Reply:
I guess it could just use a std array of uint8_t but yea I was just recommended to use the u8string instead of the original c style stuff that was there. What would be the ideal solution? I'm not sure how changing the underlying union in the widget stuff would affect things so I don't know if the solution would be in that area or not or something like I mentioned.